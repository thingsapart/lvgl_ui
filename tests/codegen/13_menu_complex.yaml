../../examples/13_menu_complex.yaml