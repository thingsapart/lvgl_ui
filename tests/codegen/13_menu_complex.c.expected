/* AUTO-GENERATED by the 'c_code' backend */

#include "lvgl.h"
#include "c_gen/lvgl_dispatch.h" // For obj_registry_add
#include "data_binding.h"

void create_ui(lv_obj_t* parent) {
    // menu: menu_0 (menu)
    lv_obj_t* menu_0 = lv_menu_create(parent);

    obj_registry_add("menu", menu_0);
    lv_obj_set_style_bg_color(menu_0, lv_color_darken(lv_obj_get_style_bg_color(menu_0, 0), 10), 0);
    lv_menu_set_mode_root_back_button(menu_0, LV_MENU_ROOT_BACK_BUTTON_ENABLED);
    lv_obj_set_style_size(menu_0, lv_pct(100), lv_pct(100), 0);
    lv_obj_center(menu_0);
    // mechanics_page: mechanics_page_1 (obj)
    lv_obj_t* mechanics_page_1 = lv_menu_page_create(menu_0, "NULL");

    obj_registry_add("mechanics_page", mechanics_page_1);
    lv_obj_set_style_pad_hor(mechanics_page_1, lv_obj_get_style_pad_left(lv_menu_get_main_header(menu_0), 0), 0);
    // unnamed: obj_2 (obj)
    lv_obj_t* obj_2 = lv_menu_separator_create(mechanics_page_1);

    // unnamed: obj_3 (obj)
    lv_obj_t* obj_3 = lv_menu_section_create(mechanics_page_1);

    // unnamed: obj_4 (obj)
    lv_obj_t* obj_4 = lv_menu_cont_create(mechanics_page_1);

    lv_obj_set_style_width(obj_4, lv_pct(100), 0);
    // unnamed: label_5 (label)
    lv_obj_t* label_5 = lv_label_create(obj_4);

    lv_label_set_text(label_5, "Velocity");
    lv_label_set_long_mode(label_5, LV_LABEL_LONG_MODE_SCROLL_CIRCULAR);
    lv_obj_set_style_flex_grow(label_5, 1, 0);

    // unnamed: label_6 (label)
    lv_obj_t* label_6 = lv_label_create(obj_4);

    lv_label_set_text(label_6, "\xef\x80\x93");
    lv_obj_add_flag(label_6, LV_OBJ_FLAG_FLEX_IN_NEW_TRACK);

    // unnamed: slider_7 (slider)
    lv_obj_t* slider_7 = lv_slider_create(obj_4);

    lv_obj_set_style_flex_grow(slider_7, 1, 0);
    lv_slider_set_range(slider_7, 0, 150);
    lv_slider_set_value(slider_7, 120, 1);


    // unnamed: obj_8 (obj)
    lv_obj_t* obj_8 = lv_menu_cont_create(mechanics_page_1);

    lv_obj_set_style_width(obj_8, lv_pct(100), 0);
    // unnamed: label_9 (label)
    lv_obj_t* label_9 = lv_label_create(obj_8);

    lv_label_set_text(label_9, "Acceleration");
    lv_label_set_long_mode(label_9, LV_LABEL_LONG_MODE_SCROLL_CIRCULAR);
    lv_obj_set_style_flex_grow(label_9, 1, 0);

    // unnamed: label_10 (label)
    lv_obj_t* label_10 = lv_label_create(obj_8);

    lv_label_set_text(label_10, "\xef\x80\x93");
    lv_obj_add_flag(label_10, LV_OBJ_FLAG_FLEX_IN_NEW_TRACK);

    // unnamed: slider_11 (slider)
    lv_obj_t* slider_11 = lv_slider_create(obj_8);

    lv_obj_set_style_flex_grow(slider_11, 1, 0);
    lv_slider_set_range(slider_11, 0, 150);
    lv_slider_set_value(slider_11, 50, 1);


    // unnamed: obj_12 (obj)
    lv_obj_t* obj_12 = lv_menu_cont_create(mechanics_page_1);

    lv_obj_set_style_width(obj_12, lv_pct(100), 0);
    // unnamed: label_13 (label)
    lv_obj_t* label_13 = lv_label_create(obj_12);

    lv_label_set_text(label_13, "Weight limit");
    lv_label_set_long_mode(label_13, LV_LABEL_LONG_MODE_SCROLL_CIRCULAR);
    lv_obj_set_style_flex_grow(label_13, 1, 0);

    // unnamed: label_14 (label)
    lv_obj_t* label_14 = lv_label_create(obj_12);

    lv_label_set_text(label_14, "\xef\x80\x93");
    lv_obj_add_flag(label_14, LV_OBJ_FLAG_FLEX_IN_NEW_TRACK);

    // unnamed: slider_15 (slider)
    lv_obj_t* slider_15 = lv_slider_create(obj_12);

    lv_obj_set_style_flex_grow(slider_15, 1, 0);
    lv_slider_set_range(slider_15, 0, 150);
    lv_slider_set_value(slider_15, 80, 1);




    // sound_page: sound_page_16 (obj)
    lv_obj_t* sound_page_16 = lv_menu_page_create(menu_0, "NULL");

    obj_registry_add("sound_page", sound_page_16);
    lv_obj_set_style_pad_hor(sound_page_16, lv_obj_get_style_pad_left(lv_menu_get_main_header(menu_0), 0), 0);
    // unnamed: obj_17 (obj)
    lv_obj_t* obj_17 = lv_menu_separator_create(sound_page_16);

    // unnamed: obj_18 (obj)
    lv_obj_t* obj_18 = lv_menu_section_create(sound_page_16);

    // unnamed: obj_19 (obj)
    lv_obj_t* obj_19 = lv_menu_cont_create(obj_18);

    // unnamed: label_20 (label)
    lv_obj_t* label_20 = lv_label_create(obj_19);

    lv_label_set_text(label_20, "\xef\x80\x81");

    // unnamed: label_21 (label)
    lv_obj_t* label_21 = lv_label_create(obj_19);

    lv_label_set_text(label_21, "Sound");
    lv_label_set_long_mode(label_21, LV_LABEL_LONG_MODE_SCROLL_CIRCULAR);
    lv_obj_set_style_flex_grow(label_21, 1, 0);

    // unnamed: switch_22 (switch)
    lv_obj_t* switch_22 = lv_switch_create(obj_19);

    lv_obj_add_state(switch_22, LV_STATE_CHECKED);




    // display_page: display_page_23 (obj)
    lv_obj_t* display_page_23 = lv_menu_page_create(menu_0, "NULL");

    obj_registry_add("display_page", display_page_23);
    lv_obj_set_style_pad_hor(display_page_23, lv_obj_get_style_pad_left(lv_menu_get_main_header(menu_0), 0), 0);
    // unnamed: obj_24 (obj)
    lv_obj_t* obj_24 = lv_menu_separator_create(display_page_23);

    // unnamed: obj_25 (obj)
    lv_obj_t* obj_25 = lv_menu_section_create(display_page_23);

    // unnamed: obj_26 (obj)
    lv_obj_t* obj_26 = lv_menu_cont_create(display_page_23);

    lv_obj_set_style_width(obj_26, lv_pct(100), 0);
    // unnamed: label_27 (label)
    lv_obj_t* label_27 = lv_label_create(obj_26);

    lv_label_set_text(label_27, "Brightness");
    lv_label_set_long_mode(label_27, LV_LABEL_LONG_MODE_SCROLL_CIRCULAR);
    lv_obj_set_style_flex_grow(label_27, 1, 0);

    // unnamed: label_28 (label)
    lv_obj_t* label_28 = lv_label_create(obj_26);

    lv_label_set_text(label_28, "\xef\x80\x93");
    lv_obj_add_flag(label_28, LV_OBJ_FLAG_FLEX_IN_NEW_TRACK);

    // unnamed: slider_29 (slider)
    lv_obj_t* slider_29 = lv_slider_create(obj_26);

    lv_obj_set_style_flex_grow(slider_29, 1, 0);
    lv_slider_set_range(slider_29, 0, 150);
    lv_slider_set_value(slider_29, 100, 1);




    // menu_page: menu_page_30 (obj)
    lv_obj_t* menu_page_30 = lv_menu_page_create(menu_0, "NULL");

    obj_registry_add("menu_page", menu_page_30);
    lv_obj_set_style_pad_hor(menu_page_30, lv_obj_get_style_pad_left(lv_menu_get_main_header(menu_0), 0), 0);
    // unnamed: obj_31 (obj)
    lv_obj_t* obj_31 = lv_menu_separator_create(menu_page_30);

    // unnamed: obj_32 (obj)
    lv_obj_t* obj_32 = lv_menu_section_create(menu_page_30);

    // unnamed: obj_33 (obj)
    lv_obj_t* obj_33 = lv_menu_cont_create(menu_page_30);

    // unnamed: label_34 (label)
    lv_obj_t* label_34 = lv_label_create(obj_33);

    lv_label_set_text(label_34, "\xef\x80\x81");

    // unnamed: label_35 (label)
    lv_obj_t* label_35 = lv_label_create(obj_33);

    lv_label_set_text(label_35, "Sidebar enable");
    lv_label_set_long_mode(label_35, LV_LABEL_LONG_MODE_SCROLL_CIRCULAR);
    lv_obj_set_style_flex_grow(label_35, 1, 0);

    // unnamed: switch_36 (switch)
    lv_obj_t* switch_36 = lv_switch_create(obj_33);

    lv_obj_add_state(switch_36, LV_STATE_CHECKED);




    // root_page: root_page_37 (obj)
    lv_obj_t* root_page_37 = lv_menu_page_create(menu_0, "Settings");

    obj_registry_add("root_page", root_page_37);
    lv_obj_set_style_pad_hor(root_page_37, lv_obj_get_style_pad_left(lv_menu_get_main_header(menu_0), 0), 0);
    // unnamed: obj_38 (obj)
    lv_obj_t* obj_38 = lv_menu_section_create(root_page_37);

    lv_obj_set_style_bg_color(obj_38, lv_color_hex(0xDDDDDD), 0);
    lv_obj_set_style_bg_opa(obj_38, 200, 0);
    // mechanics_entry: mechanics_entry_39 (obj)
    lv_obj_t* mechanics_entry_39 = lv_menu_cont_create(obj_38);

    obj_registry_add("mechanics_entry", mechanics_entry_39);
    // unnamed: obj_40 (obj)
    void obj_40;
    lv_menu_set_load_page_event(menu_0, mechanics_entry_39, mechanics_page_1);

    // unnamed: label_41 (label)
    lv_obj_t* label_41 = lv_label_create(mechanics_entry_39);

    lv_label_set_text(label_41, "\xef\x80\x93");

    // unnamed: label_42 (label)
    lv_obj_t* label_42 = lv_label_create(mechanics_entry_39);

    lv_label_set_text(label_42, "Mechanics");
    lv_label_set_long_mode(label_42, LV_LABEL_LONG_MODE_SCROLL_CIRCULAR);
    lv_obj_set_style_flex_grow(label_42, 1, 0);


    // sound_entry: sound_entry_43 (obj)
    lv_obj_t* sound_entry_43 = lv_menu_cont_create(obj_38);

    obj_registry_add("sound_entry", sound_entry_43);
    // unnamed: obj_44 (obj)
    void obj_44;
    lv_menu_set_load_page_event(menu_0, sound_entry_43, sound_page_16);

    // unnamed: label_45 (label)
    lv_obj_t* label_45 = lv_label_create(sound_entry_43);

    lv_label_set_text(label_45, "\xef\x80\x81");

    // unnamed: label_46 (label)
    lv_obj_t* label_46 = lv_label_create(sound_entry_43);

    lv_label_set_text(label_46, "Sound");
    lv_label_set_long_mode(label_46, LV_LABEL_LONG_MODE_SCROLL_CIRCULAR);
    lv_obj_set_style_flex_grow(label_46, 1, 0);


    // display_entry: display_entry_47 (obj)
    lv_obj_t* display_entry_47 = lv_menu_cont_create(obj_38);

    obj_registry_add("display_entry", display_entry_47);
    // unnamed: obj_48 (obj)
    void obj_48;
    lv_menu_set_load_page_event(menu_0, display_entry_47, display_page_23);

    // unnamed: label_49 (label)
    lv_obj_t* label_49 = lv_label_create(display_entry_47);

    lv_label_set_text(label_49, "\xef\x80\x93");

    // unnamed: label_50 (label)
    lv_obj_t* label_50 = lv_label_create(display_entry_47);

    lv_label_set_text(label_50, "Display");
    lv_label_set_long_mode(label_50, LV_LABEL_LONG_MODE_SCROLL_CIRCULAR);
    lv_obj_set_style_flex_grow(label_50, 1, 0);



    // unnamed: label_51 (label)
    lv_obj_t* label_51 = lv_label_create(root_page_37);

    lv_label_set_text(label_51, "Others");

    // unnamed: obj_52 (obj)
    lv_obj_t* obj_52 = lv_menu_section_create(root_page_37);

    // unnamed: obj_53 (obj)
    lv_obj_t* obj_53 = lv_menu_cont_create(obj_52);

    // unnamed: obj_54 (obj)
    void obj_54;
    lv_menu_set_load_page_event(menu_0, obj_53, display_page_23);

    // unnamed: label_55 (label)
    lv_obj_t* label_55 = lv_label_create(obj_53);

    lv_label_set_text(label_55, "   ");

    // unnamed: label_56 (label)
    lv_obj_t* label_56 = lv_label_create(obj_53);

    lv_label_set_text(label_56, "About");
    lv_label_set_long_mode(label_56, LV_LABEL_LONG_MODE_SCROLL_CIRCULAR);
    lv_obj_set_style_flex_grow(label_56, 1, 0);


    // unnamed: obj_57 (obj)
    lv_obj_t* obj_57 = lv_menu_cont_create(obj_52);

    // unnamed: label_58 (label)
    lv_obj_t* label_58 = lv_label_create(obj_57);

    lv_label_set_text(label_58, "\xef\x80\x93");

    // unnamed: label_59 (label)
    lv_obj_t* label_59 = lv_label_create(obj_57);

    lv_label_set_text(label_59, "Menu mode");
    lv_label_set_long_mode(label_59, LV_LABEL_LONG_MODE_SCROLL_CIRCULAR);
    lv_obj_set_style_flex_grow(label_59, 1, 0);




    lv_menu_set_sidebar_page(menu_0, root_page_37);

}
